src/main/groovy/foundation/omni/test/TestSupport.groovy

```diff
@@ -72,58 +71,70 @@ trait TestSupport implements MastercoinClientDelegate {
+
+            // TODO: can we always get away with not generating a block inbetween?
+            def extraTxidMSC = send_MP(toAddress, junkAddress, CurrencyID.MSC, excessiveMSC)
+            def extraTxidTMSC = send_MP(toAddress, junkAddress, CurrencyID.TMSC, excessiveMSC)
+        }
+
+        // TODO: when using an intermediate receiver, this txid doesn't reflect the whole picture
         return txid
     }
 
     Address createFundedAddress(BigDecimal requestedBTC, BigDecimal requestedMSC) {
-        final MPNetworkParameters params = MPRegTestParams.get()  // Hardcoded for RegTest for now
-        Address stepAddress = getNewAddress()
-        def btcForMSC = (requestedMSC / 100).setScale(8, BigDecimal.ROUND_UP)
-        def startBTC = requestedBTC + btcForMSC + stdTxFee
-        startBTC += 5 * stdTxFee  // To send BTC, MSC and TMSC to the real receiver
-
-        Sha256Hash txid = requestBitcoin(stepAddress, startBTC)
-        generateBlock()
-        def tx = getTransaction(txid)
-        assert tx.confirmations == 1
-
-        // Make sure we got the correct amount of BTC
-        BigDecimal btcBalance = getBitcoinBalance(stepAddress)
-        assert btcBalance == startBTC
-
-        // Send BTC to get MSC (and TMSC)
-        txid = sendBitcoin(stepAddress, params.moneyManAddress, btcForMSC)
-        generateBlock()
-        tx = getTransaction(txid)
-        assert tx.confirmations == 1
-
-        // Send to the actual destination
-        Address fundedAddress = getNewAddress()
-        send_MP(stepAddress, fundedAddress, CurrencyID.MSC, requestedMSC)
-        send_MP(stepAddress, fundedAddress, CurrencyID.TMSC, requestedMSC)
-        generateBlock()
-        def remainingBTC = requestedBTC - getBitcoinBalance(fundedAddress)
-        txid = sendBitcoin(stepAddress, fundedAddress, remainingBTC)
-        generateBlock()
-        tx = getTransaction(txid)
-        assert tx.confirmations == 1
-
-        // Verify correct amounts received
-        btcBalance = getBitcoinBalance(fundedAddress)
-        BigDecimal mscBalance = getbalance_MP(fundedAddress, CurrencyID.MSC).balance
-        BigDecimal tmscBalance = getbalance_MP(fundedAddress, CurrencyID.TMSC).balance
-
-        assert btcBalance == requestedBTC
-        assert mscBalance == requestedMSC
-        assert tmscBalance == requestedMSC
+        return createFundedAddress(requestedBTC, requestedMSC, true)
+    }
+
+    Address createFundedAddress(BigDecimal requestedBTC, BigDecimal requestedMSC, Boolean confirmTransactions) {
+        def fundedAddress = newAddress
+
+        if (requestedMSC > 0.0) {
+            def txidMSC = requestMSC(fundedAddress, requestedMSC)
+        }
+
+        if (requestedBTC > 0.0) {
+            def txidBTC = requestBitcoin(fundedAddress, requestedBTC)
+        }
+
+        if (confirmTransactions) {
+            generateBlock()
+        }
+
+        // TODO: maybe add assertions to check correct funding amounts?
```

Comment:
TODO: I removed the assertions, because there is no Master Core support for uncofirmed transactions, but it should be tested, whenever possible.